homeassistant/components/huawei_lte/__init__.py

```diff
@@ -492,72 +510,68 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up Huawei LTE component."""
-
     if DOMAIN not in hass.data:
         hass.data[DOMAIN] = HuaweiLteData(hass_config=config, routers={})
 
-    def service_handler(service: ServiceCall) -> None:
-        """Apply a service.
-
-        We key this using the router URL instead of its unique id / serial number,
-        because the latter is not available anywhere in the UI.
-        """
+    async def service_handler(service: ServiceCall) -> None:
+        """Apply a service."""
         routers = hass.data[DOMAIN].routers
-        if url := service.data.get(CONF_URL):
-            router = next(
-                (router for router in routers.values() if router.url == url), None
-            )
+        url = service.data.get(CONF_URL)
+        router = None
+
+        if url:
+            router = next((r for r in routers.values() if r.url == url), None)
         elif not routers:
             _LOGGER.error("%s: no routers configured", service.service)
-            return
         elif len(routers) == 1:
             router = next(iter(routers.values()))
         else:
             _LOGGER.error(
                 "%s: more than one router configured, must specify one of URLs %s",
                 service.service,
-                sorted(router.url for router in routers.values()),
+                sorted(r.url for r in routers.values()),
             )
-            return
```

Comment:
Same with this return.
Variable routers is not used after this point either, so I think these returns should be added back